packages/driver/src/cy/assertions.coffee

```diff
@@ -56,6 +58,35 @@ parseValueActualAndExpected = (value, actual, expected) ->
 
   obj
 
+prepareObjsForDiff = (err) ->
+  if _.isString(err.actual) || _.isString(err.expected)
+    return err
+  ret = {}
+  ret.actual = mochaUtils.stringify(err.actual)
+  ret.expected = mochaUtils.stringify(err.expected)
+  return ret
+
+objToString = Object.prototype.toString
+
+_isSameType = (a, b) ->
+  return objToString.call(a) is objToString.call(b)
```

Comment:
will this return same `[object Object]` for most things? Like array vs object vs null?

Reply:
yes as described here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/toString#Using_toString()_to_detect_object_class